bookkeeper-dist/src/main/resources/NOTICE-all.bin.txt

```diff
@@ -1,55 +1,136 @@
+
+Please visit the Netty web site for more information:
+
+  * http://netty.io/
+
+Copyright 2014 The Netty Project
+
+The Netty Project licenses this file to you under the Apache License,
+version 2.0 (the "License"); you may not use this file except in compliance
+with the License. You may obtain a copy of the License at:
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+License for the specific language governing permissions and limitations
 under the License.
 
-This project includes:
-  Apache Log4j under The Apache Software License, Version 2.0
-  Caffeine under The Apache Software License, Version 2.0
-  Commons CLI under The Apache Software License, Version 2.0
-  Commons Codec under The Apache Software License, Version 2.0
-  Commons Collections under The Apache Software License, Version 2.0
-  Commons Configuration under The Apache Software License, Version 2.0
-  Commons IO under The Apache Software License, Version 2.0
-  Commons Lang under The Apache Software License, Version 2.0
-  Commons Logging under The Apache Software License, Version 2.0
-  Dropwizard Metrics under The Apache Software License, Version 2.0
-  Guava under The Apache Software License, Version 2.0
-  Jackson under The Apache Software License, Version 2.0
-  Java Native Access under The Apache Software License, Version 2.0
-  Java Servlet API under Common Development and Distribution License 1.0
-  Javax.inject under The Apache Software License, Version 2.0
-  Jetty under The Apache Software License, Version 2.0
-  JSR305 Annotations for Findbugs under The Apache Software License, Version 2.0
-  ParaNamer under New BSD License
-  Prometheus under The Apache Software License, Version 2.0
-  Protocol Buffer Java API under New BSD License
-  Scala under New BSD License
-  SLF4J API Module under MIT License
-  SLF4J LOG4J-12 Binding under MIT License
-  The Netty Project under The Apache Software License, Version 2.0
-  Twitter Finagle under The Apache Software License, Version 2.0
-  Twitter JSR166e under Creative Commons Zero v1.0 Universal
-  Twitter LibThrift under The Apache Software License, Version 2.0
-  Twitter Server under The Apache Software License, Version 2.0
-  Twitter Scrooge under The Apache Software License, Version 2.0
-  Twitter Util under The Apache Software License, Version 2.0
-  Vertx under The Apache Software License, Version 2.0
-  ZooKeeper under The Apache Software License, Version 2.0
-
-This product contains a modified version of 'Circe', a high-performance
-hash algorithm framework & library from Trevor Robinson:
-
-  * LICENSE: Apache License 2.0
-  * HOMEPAGE: https://github.com/trevorr/circe
+lib/io.netty-netty-all-4.1.12.Final.jar contains a modified portion of 'Apache Harmony', an open source
```

Comment:
where does this come from? 

1) I didn't see it appears in netty's notice: https://github.com/netty/netty/blob/netty-4.1.12.Final/NOTICE.txt#L61
2) it doesn't belong to relocated copyright. relocated copyright only applies for src files. http://www.apache.org/legal/src-headers.html#headers

Reply:
From the harmony NOTICE.
http://svn.apache.org/viewvc/harmony/enhanced/java/trunk/NOTICE?revision=929253&view=markup

The bit they pull from harmony is parts of java.lang.String, which is from intel.
http://svn.apache.org/viewvc/harmony/enhanced/classlibadapter/trunk/modules/kernel/src/main/java/java/lang/String.java?view=markup#l88

netty should be bubbling this up.

Reply:
https://github.com/netty/netty/commit/681d4609382f9741325b0b2743eb19b85352f425

Reply:
@ivankelly netty is not an ASF project and it is not our responsibility for modifying a NOTICE file beyond projects. we are bundled netty 4.1.12, which should use the NOTICE in 4.1.12, if netty is going to fix that in future releases and we bump the version, we include the updated NOTICE.

Reply:
Apache harmony is an ASF project. netty doesn't include any NOTICE for harmony, and this is wrong. We ship code which belongs to harmony, so we should also ship it's notice. That it come in via netty is largely irrelevant. We can ask about this to legal-discuss@ if you like.

Reply:
please ask. we are not directly bundling harmony, we bundled netty. harmony is not the dependency of netty, it is the source code copied by netty. I don't think we need to include harmony NOTICE, because as what you said NOTICE can be changed between versions, you can't use the latest NOTICE in harmony as the NOTICE when netty introduced the modified code. so we have to stick to whatever NOTICE shipped by netty.

Reply:
I'll email them tomorrow

Reply:
I dug down deeper. The String class that netty copies from is originally from IBM,
https://issues.apache.org/jira/browse/HARMONY-14
http://mail-archives.apache.org/mod_mbox/harmony-dev/200511.mbox/browser

They've assigned ASF copyright in the initial contrib though, except for ICU stuff, which isn't part of String, so it turns out there's no mention needed in the notice. Will remove from this patch. Netty NOTICE is still wrong though, will ping them about it.